analytic_engine/src/table_meta_set_impl.rs

```diff
@@ -215,6 +218,15 @@ impl TableMetaSetImpl {
                 msg: format!("space not found, space_id:{space_id}"),
             })?;
 
+        // Apply max file id to the allocator
+        let allocator = match version_meta.clone() {
+            Some(version_meta) => {
+                let max_file_id = version_meta.max_file_id_to_add();
```

Comment:
I guess the `max_file_id` can only be used when the `max_file_id` aligned by the step is not found, which is actually an issue about the compatibility.